File: development/scripts/postinstall.py
import os
import shutil

def postinstall():
    # Run yarn setup:env
    print("Running yarn setup:env...")
    os.system("yarn setup:env")

    # Run patch-package
    print("Running patch-package...")
    os.system("patch-package")

    # Run yarn copy:inject
    print("Running yarn copy:inject...")
    os.system("yarn copy:inject")

    # Remove realm-flipper-plugin-device src directory
    realm_path = "node_modules/realm-flipper-plugin-device/src"
    if os.path.exists(realm_path):
        print(f"Removing directory: {realm_path}")
        shutil.rmtree(realm_path)

if __name__ == "__main__":
    postinstall()
    print("Post-installation steps completed.")

File: development/scripts/postinstall.js
#!/usr/bin/env node

const { execFileSync, execSync } = require('child_process');
const path = require('path');

console.log('Running postinstall script...');

// Run setup:env, patch-package, and copy:inject
execSync('yarn setup:env && patch-package && yarn copy:inject', {
  stdio: 'inherit',
});

if (process.platform === 'darwin') {
  // patch-package runs after dependency install, so rebuild Noble from the
  // patched Objective-C++ source instead of loading its bundled prebuild.
  const electronRebuildCli = path.join(
    path.dirname(require.resolve('@electron/rebuild')),
    'cli.js',
  );
  execFileSync(
    process.execPath,
    [
      electronRebuildCli,
      '--force',
      '--build-from-source',
      '--module-dir',
      'apps/desktop',
      // @electron/rebuild matches scoped packages by their basename here.
      '--only',
      'noble',
    ],
    { stdio: 'inherit' },
  );
}

console.log('Postinstall script completed.');
